tests/containers/podmansh.pm

```diff
@@ -0,0 +1,200 @@
+    systemctl("--machine=$username@.host --user start podmansh");
+    systemctl("--machine=$username@.host --user is-active podmansh");
+
+    # change the dafault podmansh non-root shell to match 'text-logged-in-bernhard' needle
+    assert_script_run(qq(sudo -su $user sh -c 'cd; mkdir -p ~/.config/containers/; printf '\\''[podmansh]\\nshell = \"/bin/bash\"\\n'\\'' > ~/.config/containers/containers.conf'));
+
+    # validate podmansh container for multiple login methods
+    my @cmds = (
+        "sudo -iu $user sh -c",
+        "su --login $user -c",
+        "ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -q $user\@localhost"
+    );
+    for my $cmd (@cmds) {
+        execute_tests($cmd);
+    }
+
+    # Remove all previous commands generated by root. Some of these commands will be triggered
+    # by the rootless user and will generate the same file /tmp/scriptX which will fail if it
+    # already exists owned by root
+    assert_script_run 'rm -rf /tmp/script*';
+    ensure_serialdev_permissions;
+    select_console "user-console";
+
+    execute_tests("sh -c");
+}
+
+sub execute_tests {
+    my $cmd = shift;
+    validate_script_output("$cmd id", sub { /^uid=1000\(bernhard\) gid=1000\(bernhard\) groups=1000\(bernhard\)$/ });
+    validate_script_output("$cmd 'cat /etc/os-release'", sub { /^VERSION=\"15\.6\"$/m && /^ID=\"opensuse-leap\"$/m });
+}
+
+sub prepare_rootless {
+    my $self = shift;
+    my $user = $testapi::username;
+
+    # --- START borrowed from rootless_podman.pm ---
+    # TODO: IS THIS REQUIRED?
+    if (get_var('TDUP')) {
+        my $cont_storage = '/etc/containers/storage.conf';
+        my $local_storage = "/home/$user/.local";
+
+        # use only packaged storage config in /usr/share/containers/storage.conf
+        if (script_run("test -f $cont_storage &> /dev/null") == 0) {
+            assert_script_run("rm -rf $cont_storage");
+            assert_script_run('rm -rf /var/lib/containers/storage');
+        }
+
+        # prevent database graph driver mismatch
+        if (script_run("test -d $local_storage &> /dev/null") == 0) {
+            record_soft_failure('gh#containers/podman#24738');
+            assert_script_run("rm -rf $local_storage");
+        }
+
+        assert_script_run('podman system reset -f');
+    }
+
+    # Some products don't have bernhard pre-defined (e.g. SLE Micro)
+    if (script_run("grep $user /etc/passwd") != 0) {
+        assert_script_run "useradd -m $user";
+        assert_script_run "echo '$user:$testapi::password' | chpasswd";
+        # Make sure user has access to tty group
+        my $serial_group = script_output "stat -c %G /dev/$testapi::serialdev";
+        assert_script_run "grep '^${serial_group}:.*:${user}\$' /etc/group || (chown $user /dev/$testapi::serialdev && gpasswd -a $user $serial_group)";
+    }
+    # --- END borrowed from rootless_podman.pm ---
+}
+
+sub pre_run_hook {
+    $user = $testapi::username;
```

Comment:
Given that you modify some crucial things of the global testuser, perhaps it makes more sense to add a dedicated `podmansh` user. This avoids conflicting configurations in other places.